digitalfabrik/entitlementcard
# 1400: Activate card confirmation mail per region

### Short description

As a region i may not want to make the card earlier usable and not send an activation mail.

### Proposed changes


- adjust activated for application to region specific settings and add checkbox for activate conformation mail
- also adjusted the query and update functions to not have x single functions for each setting
- send only card confirmation mail if its activated
- set the confirmationFeatureFlag true
- add unit test
- added jest-dom and types to enable checkbox checking in tests

### Resolved issues



Fixes: #1400 

### Testing
- run db migration (dev only)
- login as region admin (bavaria)
- go to region
- set checkbox activate confirmation mail
- create card from application
- toast 'Benachrichtigung wurde gesendet'  should be shown
- mail should be sent
- uncheck the checkbox in region settings
- create card from application
- mail shouldn't be send

**Notes:** Many changes are just because region data class has changed
